### PR TITLE
Initial refactor for Cycle Unified

### PR DESCRIPTION
**Pull request by EnigmaCurry**
I'm following the Cycle.js Unified [release notes for updating drivers](https://cycle.js.org/releases.html#cycle-unified-for-library-authors-driver-authors) and I've made some initial refactorings for cyclic-router.

Still getting some test errors, so please don't merge this yet, but I wanted to share what I've got so far. I'm still investigating, so please let me know of any tips.

### CODE REVIEW

**Comment by EnigmaCurry** [comment]
@TylorS the only tests still failing are timeouts. If you can give me any pointers, I'll keep hacking on this.

File: test/xstream.js [hunk 1]
```diff
@@ -139,8 +139,8 @@ describe('Cyclic Router - XStream', () => {
         },
       }
 
-      const history = createServerHistory('/some/route')
-      const router = makeRouterDriver(history, switchPath)(xs.never(), XSAdapter)
+      const history = createMemoryHistory('/some/route')
+      const router = makeRouterDriver(history, switchPath)(xs.never())
```
Comment (EnigmaCurry) [inline test/xstream.js]: Changing this to 

    makeRouterDriver(history, switchPath)(xs.of('/some/route'))

makes the test pass, I'm really not understanding this quite well enough to know if that makes the test invalid or not though?

Comment (TylorS) [inline test/xstream.js]: I think it works something like `createMemoryHistory({ initialPaths: [ '/some/route' ] })`

**Comment by TylorS** [comment]
I don't really have the time to maintain this library, hence #183, if you're interested please let me know there. Otherwise, if you're not interested, just let me know when this is ready to merge and I'll do a major release.

**Comment by EnigmaCurry** [comment]
OK, that's cool. If I can figure out these test failures I'll let you know about helping with maintenance.

**Comment by EnigmaCurry** [comment]
I'm closing this, because I couldn't figure out my issues. The single file router from https://github.com/garrydzeng/cycle-page/ was easier for me to grok, although it's probably a lot less powerful than this. Sorry, hope someone else can pick up where I left off. Thanks again for all you do Tylor, your work is much appreciated!

**Comment by ntilwalli** [comment]
@EnigmaCurry @TylorS See https://github.com/cyclejs-community/cyclic-router/issues/185

**Pull request closed without merging**